File: exercicios/exercicio1php/Produto.php
<?php
class Produto {
    private $nome;
    private $preco;
    private $quantidade;
    private $desconto;
    private $valorFinal;
    
    //$nome
    public function getNome() {
        return $this->nome;
    }
    public function __contruct($nome) {
        $this->$nome = $nome;
    }
    //preço
    public function getPreco($preco,$quantidade) {
        $this->$preco= $preco*$quantidade;
        return $this->preco;
    }
    public function getPrecoInicial(){
        return $this->preco;
    }
    public function __construct($preco) {
        $this->$preco = $preco;
    }
    //quantidade
    public function getQuantidade() {
        return $this->quantidade;
    }
    public function __construct($quantidade) {
        $this->$quantidade = $quantidade;
    }
    //$desconto
    public function getDesconto() {
        $this->desconto = 0;
        if ($quantidade>10 and $quantidade<=20)
            $desconto = ($preco)*10/100;
        else if ($quantidade>20 and $quantidade<=50)
            $desconto = ($preco)*20/100;
        else if ($quantidade>50)
            $desconto = ($preco)*25/100;
        
        return $desconto;
    }
    public function setDesconto($desconto) {
        $this->desconto = $desconto;
    }
    //valorFinal
    public function getValorFinal() {
        $valorFinal = $preco-$desconto;
        return $valorFinal;
    }
    public function setValorFinal($valorFinal) {
        $this->$valorFinal = $valorFinal;
    }
    public function __construct($nome, $preco, $quantidade) {
        $this->nome = $nome;
        $this->preco = $preco*$quantidade;
        $this->quantidade = $quantidade;
        $desconto = getDesconto();
    }
}
?>

File: exercicios/exercicio1java/Produto.java
package exercicios.exercicio1java;

public class Produto {
    private String nome;
    private double preco;
    private int quantidade;
    private double desconto;
    private double valorFinal;
    
    //nome
    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }
    //preço
    public double getPreco() {
        preco = preco*quantidade;
        return preco;
    }
    public double getPrecoInicial(){
        return preco;
    }
    public void setPreco(double preco) {
        this.preco = preco;
    }
    //quantidade
    public int getQuantidade() {
        return quantidade;
    }
    public void setQuantidade(int quantidade) {
        this.quantidade = quantidade;
    }
    //desconto
    public double getDesconto() {
        desconto = 0;
        if (quantidade>10 && quantidade<=20)
            desconto = (preco)*10/100;
        else if (quantidade>20 && quantidade<=50)
            desconto = (preco)*20/100;
        else if (quantidade>50)
            desconto = (preco)*25/100;
        
        return desconto;
    }
    public void setDesconto(double desconto) {
        this.desconto = desconto;
    }
    //valorFinal
    public double getValorFinal() {
        valorFinal = preco-desconto;
        return valorFinal;
    }
    public void setValorFinal(double valorFinal) {
        this.valorFinal = valorFinal;
    }
    Produto(String nome, double preco, int quantidade) {
        this.nome = nome;
        this.preco = preco*quantidade;
        this.quantidade = quantidade;
        desconto = getDesconto();
    }
    Produto(){}
}
